packages/components/src/components/FileUploader/FileUploader.tsx

```diff
@@ -63,13 +66,18 @@ const FileUploader = React.forwardRef<HTMLDivElement, FileUploaderProps>(
       onCancel,
       onRemove,
       disabled = false,
+      required,
     },
     ref
   ) => {
     const isMobile = useDown("md");
     const status = getStatus({ progress, fileUrl, errorMessage });
     const isMobileUploading = isMobile && status === "loading";
     const isUploadedWithoutName = status === "success" && !fileName;
+    const hasFile = fileName || fileUrl;
+    const shouldShowChildren = status === "waiting" || (required && !hasFile);
```

Comment:
If there's an error but there's no file, then it wouldn't show children or the remove button
```suggestion
    const shouldShowChildren = status === "waiting" || (status === "error" && !hasFile);
```

Reply:
oh right! In this case we wouldn't need the `required` prop 🤔

Reply:
Yeah, I don't think the FileUploader component really needs to care about `required`. It just needs to be able to show the errors properly.